### DISCUSSION TITLE
MIME type globs in FileChooser portals?

### DISCUSSION THREAD

**Discussion by nicolasfella** [flatpak/xdg-desktop-portal#1487] [Questions]
Some applications use wildcard patterns like video/* in the MIME type filters for the FileChooser portal. 

For example Footage sends
```
dict entry(
    string "filters"
    variant             array [
        struct {
            string "Videodateien"
            array [
                struct {
                uint32 1
                string "video/*"
                }
            ]
        }
    ]
)
```
xdg-desktop-portal-kde doesn't handle those (https://bugs.kde.org/show_bug.cgi?id=450612). Before I go and fix this I wanted to ask whether portal implementations are actually expected to handle those. The interface description doesn't mention this, so it would be good to have this clarified one way or the other

**Comment by Mikenux** [comment]
It seems it isn't expected.

According to https://github.com/flatpak/xdg-desktop-portal/blob/main/data/org.freedesktop.impl.portal.FileChooser.xml, you need to look at https://github.com/flatpak/xdg-desktop-portal/blob/main/data/org.freedesktop.portal.FileChooser.xml, which says:

>     List of serialized file filters.
> 
>         Each item in the array specifies a single filter to offer to the user.
> 
>         The first string is a user-visible name for the filter. The ``a(us)``
>         specifies a list of filter strings, which can be either a glob-style
>         pattern (indicated by 0) or a MIME type (indicated by 1). Patterns are
>         case-sensitive.
> 
>         To match different capitalizations of, e.g. ``'*.ico'``, use a pattern
>         like: ``'*.[iI][cC][oO]'``.
> 
>         Example: ``[('Images', [(0, '*.ico'), (1, 'image/png')]), ('Text', [(0, '*.txt')])]``

**Comment by swick** [comment]
FWIW, the frontend does not validate the filters (and I'm not sure to what degree we even could). That means the filters go straight to the impl, which in the case of GNOME is Nautilus, which does support the MIME type globs.

Maybe we should just add it to the spec?

**Reply by Sodivad** [reply]
I think it's a sensible addition
